jujugui/static/gui/src/app/components/user-profile/user-profile.js

```diff
@@ -82,6 +92,89 @@ YUI.add('user-profile', function() {
     },
 
     /**
+      Requests a list from charmstore of the user's entities.
+
+      @method _fetchEntities
+      @param {String} type the entity type, charm or bundle
+    */
+    _fetchEntities:  function(type) {
+      var callback = this._fetchEntitiesCallback.bind(this, type);
+      var charmstore = this.props.charmstore;
+      var username = this.props.username;
+      if (charmstore && charmstore.list) {
+        charmstore.list(username, callback, type);
+      }
+    },
+
+    /**
+      Munge a single entity into data ready to be displayed on the user's
+      profile.
+
+      @method _processEntity
+      @param {Object} entity A charm or a bundle from the API.
+    */
+    _processEntity: function(entity) {
+      var tags = entity.tags || [];
+      tags = tags.map(function(tag) {
+        return (
+          <span
+            key={tag}
+            className="user-profile__entityList-tags">
+            {tag}
+          </span>
+        );
+      });
+      var cs = this.props.charmstore;
+      var url = (cs && cs.url && cs.version) && cs.url + cs.version;
```

Comment:
Is it even possible that there is no charmstore provided? This might be overly complicated for a case which isn't supported.